### ISSUE TITLE
sort + indexing => select_nth_unstable

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#14119]
### What it does

This pedantic lint is supposed to suggest replacing a sort or sort_unstable followed by an indexing, with the usage of a more efficient select_nth_unstable. I am not sure if this lint is too much specialized, if you think it's it, then close down this issue.

### Advantage

Faster, slightly more clear purpose of the code.

### Drawbacks

It's hard to spot the cases where this lint should fire. So better to keep it conservative and avoid most false positives.

### Example

```rust
#[must_use]
pub fn compute_median(data: &[u32]) -> u32 {
    let mut data_sorted = data.to_owned();
    data_sorted.sort_unstable();
    data_sorted[data_sorted.len() / 2]
}
```

The link should suggest to use instead:

```rust
select_nth_unstable()
```
